server.py

```diff
@@ -67,16 +69,23 @@ def rankings():
 def file_upload(number):
     u_name = request.forms.get("username")  # accepting username
     time = datetime.datetime.now()
-    # type(uploaded) == <class 'bytes'>
-    # uploaded outputs by user
     uploaded = request.files.get("upload").file.read()
     expected = questions[number].output
     expected = expected.strip()
     uploaded = uploaded.strip()
-    ans = uploaded == expected
-    submission_record[u_name].append(
+    ans = (uploaded == expected)
+    
+    submission_record = shelve.open("submission_record.db")
+    if not u_name in submission_record.keys():
+        submissions = []
+    else:
```

Comment:
this can actually be done like this to make it much cleaner:

```python
submissions = [] if u_name not in submission_record else submission_record[u_name]
submissions = submission_record.get(u_name, list())
```

Reply:
Is the second line alternative to first?

Reply:
yes. all of the things do the same thing but some are more easily readable